TST: 1 test failure for release 3.4.0 on Fedora 37 and 36.

#### Code Sample, a copy-pastable example if possible

Yesterdays 3.4.0 release builds and tests just fine in Fedora rawhide (future fedora 38).
However, the test suite produces one test failure when run with: py.test-3 -m "not network" for fedora 37 and 36.
I can suppress this test by using "not network and not grid", but if possible, I think it would be better to understand what is going on and to fix it if possible

The output for the failing test is:
```
__________________ test_transformer_group__get_transform_crs ___________________
    @pytest.mark.grid
    def test_transformer_group__get_transform_crs():
        tg = TransformerGroup("epsg:4258", "epsg:7415")
        if not grids_available("nl_nsgi_rdtrans2018.tif"):
>           assert len(tg.transformers) == 1
E           assert 2 == 1
E            +  where 2 = len([<Concatenated Operation Transformer: pipeline>\nDescription: Inverse of Transformation from NAP height to ETRS89 (ball...geographic offset from Amersfoort to ETRS89 + RD New\nArea of Use:\n- name: World\n- bounds: (-180.0, -90.0, 180.0, 90.0)])
E            +    where [<Concatenated Operation Transformer: pipeline>\nDescription: Inverse of Transformation from NAP height to ETRS89 (ball...geographic offset from Amersfoort to ETRS89 + RD New\nArea of Use:\n- name: World\n- bounds: (-180.0, -90.0, 180.0, 90.0)] = <TransformerGroup: best_available=False>\n- transformers: 2\n- unavailable_operations: 4.transformers
test/test_transformer.py:699: AssertionError 
```
Version details for fedora 37 are:
pyproj release 3.4.0
proj version 9.0.1
python3 version 3.11.0~rc1
gcc version  12.2.1
python3-Cython version 0.29.32

Version details for fedora rawhide/38 are:
pyproj release 3.4.0
proj version 9.1.0
python3 version 3.11.0~rc1
gcc version  12.2.1
python3-Cython version  0.29.32

The only difference seems to be the version of the proj library.

For more details see:
https://koji.fedoraproject.org/koji/buildinfo?buildID=2061414 (rawhide build)
https://koji.fedoraproject.org/koji/taskinfo?taskID=91901918 (fedora 37 build)

--- a/test/test_transformer.py
+++ b/test/test_transformer.py
@@ -695,12 +695,19 @@ def test_transform_group__area_of_interest():
 @pytest.mark.grid
 def test_transformer_group__get_transform_crs():
     tg = TransformerGroup("epsg:4258", "epsg:7415")
-    if not grids_available("nl_nsgi_rdtrans2018.tif"):
-        assert len(tg.transformers) == 1
-    elif PROJ_GTE_91:
+    if grids_available(
+        "nl_nsgi_nlgeo2018.tif", "nl_nsgi_rdtrans2018.tif", check_all=True
+    ):
+        if PROJ_GTE_91:
+            assert len(tg.transformers) == 2
+        else:
+            assert len(tg.transformers) == 6
+    elif not PROJ_GTE_91 and grids_available("nl_nsgi_rdtrans2018.tif"):
         assert len(tg.transformers) == 2
+    elif not PROJ_GTE_91 and grids_available("nl_nsgi_nlgeo2018.tif"):
+        assert len(tg.transformers) == 4
     else:
-        assert len(tg.transformers) == 6
+        assert len(tg.transformers) == 1
 
 
 def test_transformer__area_of_interest():

